### PR TITLE
Some cleanup for stateManager

### PR DESCRIPTION
**Pull request by axic**
Depends on #264. Part of #33.

### CODE REVIEW

File: lib/stateManager.js [hunk 1]
```diff
@@ -11,28 +11,22 @@ const rlp = utils.rlp
...
-  if (!blockchain) {
-    blockchain = fakeBlockchain
-  }
-
-  self.blockchain = blockchain
-  self.trie = trie
+  self.blockchain = opts.blockchain || fakeBlockchain
+  self.trie = opts.trie || new Trie()
   self._storageTries = {} // the storage trie cache
   self.cache = new Cache(trie)
```
Comment (holgerd77) [inline lib/stateManager.js]: Should ``cache`` also be made an option, so that people can easier switch that if desired?

Comment (axic) [inline lib/stateManager.js]: Should use `self.trie` here, wonder why lint didn't catch this.

File: lib/stateManager.js [hunk 2]
```diff
@@ -33,6 +33,10 @@ function StateManager (opts) {
 
 var proto = StateManager.prototype
 
+proto.clone = function () {
```
[the remark this answers is not in the document]
Comment (jwasinger) [reply lib/stateManager.js]: Yeah.  to improve consistency.

**Comment by jwasinger** [comment]
Looks good overall.  I approve of this refactor.

File: lib/stateManager.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (axic) [reply lib/stateManager.js]: Though I think `copy` is wrong overall, should be `clone`, yet we can do that in a separate step.

File: lib/stateManager.js (hunk 1 above)
Comment (axic) [reply lib/stateManager.js]: Yes, but I'd do it in a separate work. `_storageTries` is another cache too, so not sure how it works yet.

**Comment by axic** [comment]
Need to figure out why this is failing.

**Comment by holgerd77** [comment]
Just rebased this.

**Comment by holgerd77** [comment]
This is breaking at ``modexpRandomInput`` with:

```shell
# file: modexpRandomInput test: modexpRandomInput
ok 1472 the state roots should match
ok 1473 the state roots should match
/Users/hdrewes/Documents/DEV/EthereumJS/ethereumjs-vm/node_modules/rustbn.js/lib/index.asm.js:1
(function (exports, require, module, __filename, __dirname) { var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=false;var ENVIRONMENT_IS_WORKER=false;var ENVIRONMENT_IS_NODE=false;var ENVIRONMENT_IS_SHELL=false;if(Module["ENVIRONMENT"]){if(Module["ENVIRONMENT"]==="WEB"){ENVIRONMENT_IS_WEB=true}else if(Module["ENVIRONMENT"]==="WORKER"){ENVIRONMENT_IS_WORKER=true}else if(Module["ENVIRONMENT"]==="NODE"){ENVIRONMENT_IS_NODE=true}else if(Module["ENVIRONMENT"]==="SHELL"){ENVIRONMENT_IS_SHELL=true}else{throw new Error("The provided Module['ENVIRONMENT'] value is not valid. It must be one of: WEB|WORKER|NODE|SHELL.")}}else{ENVIRONMENT_IS_WEB=typeof window==="object";ENVIRONMENT_IS_WORKER=typeof importScripts==="function";ENVIRONMENT_IS_NODE=typeof process==="object

Error: Number can only safely store up to 53 bits
    at assert (/Users/hdrewes/Documents/DEV/EthereumJS/ethereumjs-vm/node_modules/bn.js/lib/bn.js:6:21)
    at BN.toNumber (/Users/hdrewes/Documents/DEV/EthereumJS/ethereumjs-vm/node_modules/bn.js/lib/bn.js:506:7)
    at getAdjustedExponentLength (/Users/hdrewes/Documents/DEV/EthereumJS/ethereumjs-vm/dist/precompiled/05-modexp.js:30:43)
    at Object.module.exports [as code] (/Users/hdrewes/Documents/DEV/EthereumJS/ethereumjs-vm/dist/precompiled/05-modexp.js:82:22)
```

Test can be run with:

```shell
node tests/tester.js -s --test='modexpRandomInput'
```

**Comment by holgerd77** [comment]
Just rebased this.

**Comment by axic** [comment]
@holgerd77 this works now! 😉

**Comment by axic** [comment]
@holgerd77 mergemerge? :)

**Pull request merged**